### PR TITLE
[Hub Generated] Review request for Microsoft.DataProtection to add version preview/2022-11-01-preview

### PR DESCRIPTION
**Pull request by anjorsh**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/anjorsh/azure-rest-api-specs/dev-dataprotection-Microsoft.DataProtection-2022-11-01-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by rkmanda** [comment]
>   "schemes": [

There are a bunch of breaking changes reported on this PR. Please get a breakingchanges approval and ARM can review the changes afterwards. Once you have the approval please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers



---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:11 in df7de47. [](commit_id = df7de47795158e4280586e2d8cd44b1c37af3da0, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "readOnly": true

This looks like a breaking change - writable to readonly. But I'm imagining its probably actually an authoring oversight in the previous swagger version?

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4868 in e6f6e6c. [](commit_id = e6f6e6cd491bcbef6f88b44633323ed3d679a8de, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Flag to specify if the path is relative to Backup Item or full path",

nit, don't capitalize - try to use plain English in docs

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:6025 in e6f6e6c. [](commit_id = e6f6e6cd491bcbef6f88b44633323ed3d679a8de, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "list of prefixes to be used relative to the ItemPath for Restore criteria",

nit, don't capitalize - e.g. just call it 'the item path for restore criteria' or refer to properties by a name with quotes if you really want to 'itemPath'

'for restore criteria' may be redundant in this context?

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:6029 in e6f6e6c. [](commit_id = e6f6e6cd491bcbef6f88b44633323ed3d679a8de, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "isPathRelativeToBackupItem": {

booleans aren't extensible

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:6024 in e6f6e6c. [](commit_id = e6f6e6cd491bcbef6f88b44633323ed3d679a8de, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "description": "Parameters for Backup",

capitalization. overly brief?

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4560 in e6f6e6c. [](commit_id = e6f6e6cd491bcbef6f88b44633323ed3d679a8de, deletion_comment = False)

**Review by TimLovellSmith — CHANGES_REQUESTED** [review]
:clock1:

**Comment by anjorsh** [comment]
> > "schemes": [
> 
> There are a bunch of breaking changes reported on this PR. Please get a breakingchanges approval and ARM can review the changes afterwards. Once you have the approval please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers
> 
> Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:11 in df7de47. [](commit_id = df7de47795158e4280586e2d8cd44b1c37af3da0, deletion_comment = False)

Breaking change approval is added to the PR now.

**Comment by anjorsh** [comment]
> > ```
> >       "readOnly": true
> > ```
> 
> This looks like a breaking change - writable to readonly. But I'm imagining its probably actually an authoring oversight in the previous swagger version?
> 
> Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4868 in [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de). [](commit_id = [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de), deletion_comment = False)

Yes, this was an oversight. This was fixed in previous Swagger version as well. 
https://github.com/Azure/azure-rest-api-specs/pull/21452/files

**Comment by anjorsh** [comment]
> > ```
> >   "description": "Parameters for Backup",
> > ```
> 
> capitalization. overly brief?
> 
> Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4560 in [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de). [](commit_id = [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de), deletion_comment = False)

Updated the description

**Comment by anjorsh** [comment]
> > ```
> >       "description": "list of prefixes to be used relative to the ItemPath for Restore criteria",
> > ```
> 
> nit, don't capitalize - e.g. just call it 'the item path for restore criteria' or refer to properties by a name with quotes if you really want to 'itemPath'
> 
> 'for restore criteria' may be redundant in this context?
> 
> Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:6029 in [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de). [](commit_id = [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de), deletion_comment = False)

Updated the description as per feedback

**Comment by anjorsh** [comment]
> > ```
> >     "isPathRelativeToBackupItem": {
> > ```
> 
> booleans aren't extensible
> 
> Replace boolean/switch properties with better enum A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].
> 
> Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:6024 in [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de). [](commit_id = [e6f6e6c](https://github.com/Azure/azure-rest-api-specs/commit/e6f6e6cd491bcbef6f88b44633323ed3d679a8de), deletion_comment = False)

This is great feedback, and we'll certainly keep it in mind during subsequent development. However, this specific contract has dependencies across multiple microservices, and it would be onerous effort to update them all at this point. Furthermore, we do not anticipate extending this property further in future. Given these issues, could this contract please be approved?

**Comment by hiaga** [comment]
@anjorsh :  I see completeness issues  [Swagger KPI Report - Power BI](https://msit.powerbi.com/groups/de995b1a-16c1-4cc7-a49b-bc9f90bc6acd/reports/17c7b280-b27a-4f7f-b24f-f23f80e7ee3e/ReportSectiond8d202227cfea6b2aa32?experience=power-bi) for API version 2022-11-01-preview

could you please check and resolving before merging this PR

**Comment by anjorsh** [comment]
> 🕐

ARM signoff was obtained.

**Comment by akning-ms** [comment]
.Net SDK generation fail, @fengzhou-msft to help take a look

**Review by xiaoxuqi-ms — APPROVED** [review]

File: custom-words.txt
Comment (TimLovellSmith) [inline custom-words.txt:2145]: Don't add this - timedout is not a single word. Its two words 'timed out'.

Comment (TimLovellSmith) [reply custom-words.txt:2145]: We can avoid adding this if we fix capitalization elsewhere

**Comment by TimLovellSmith** [comment]
>             "PrepareTimedout",

should be PrepareTimedOut

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4835 in d0a0048. [](commit_id = d0a00482813dba2dc561daf76e20be63101b4003, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "CommitTimedout",

should be CommitTimedOut

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4836 in d0a0048. [](commit_id = d0a00482813dba2dc561daf76e20be63101b4003, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "CommitTimedout",

If you've already shipped a non-preview probably this is too breaking.

---
In reply to: [1353482063](https://github.com/Azure/azure-rest-api-specs/pull/21441#issuecomment-1353482063) [](http://example.com/codeflow?ancestors=1353482063)

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4836 in 47531fc. [](commit_id = 47531fcea25ebe8b3f822b7155a89b68fa48fc17, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "PrepareTimedout",

If you've already shipped a non-preview probably this is too breaking.

---
In reply to: [1353481730](https://github.com/Azure/azure-rest-api-specs/pull/21441#issuecomment-1353481730) [](http://example.com/codeflow?ancestors=1353481730)

---
Refers to: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2022-11-01-preview/dataprotection.json:4835 in 47531fc. [](commit_id = 47531fcea25ebe8b3f822b7155a89b68fa48fc17, deletion_comment = False)

File: custom-words.txt
Comment (TimLovellSmith) [reply custom-words.txt:2145]: Looks like that would be breaking, so instead you should remove it here and suppress it with  cSpell.json

Comment (anjorsh) [reply custom-words.txt]: Removed "timedout" from custom-words.txt and added a suppression in cSpell.json instead

**Review by TimLovellSmith — APPROVED** [review]

**Comment by TimLovellSmith** [comment]
@anjorsh Yup, LGTM

**Pull request merged**
